From ca92299423980a9be5560f6eb29ba949356d638f Mon Sep 17 00:00:00 2001
From: Chris Bagwell <chris@cnpbagwell.com>
Date: Sat, 25 Jun 2011 13:05:52 -0500
Subject: [PATCH] Default to touch tool for multi-touch devices

We really need to add hooks to query pre-existing device state when
input driver starts up because slot ID is almost never sent when
touching 1 finger to screen.

In mean time, it makes sense to default device_type to most likely
value of TOUCH (the T in MT) for this function used only by MT
devices.

The default of zero is causing some events to be filtered mistakenly.

Signed-off-by: Chris Bagwell <chris@cnpbagwell.com>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/wcmUSB.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/wcmUSB.c b/src/wcmUSB.c
index 30f7c07..6ce2347 100644
--- a/src/wcmUSB.c
+++ b/src/wcmUSB.c
@@ -1371,7 +1371,7 @@ static int usbParseBTNEvent(WacomCommonPtr common,
  */
 static int usbInitToolType(const struct input_event *event_ptr, int nevents)
 {
-	int i, device_type = 0;
+	int i, device_type = TOUCH_ID;
 	struct input_event* event = (struct input_event *)event_ptr;
 
 	for (i = 0; (i < nevents) && !device_type; ++i)
-- 
1.7.0.1

